src/NMEA/EngineState.hpp

```diff
@@ -0,0 +1,62 @@
+// SPDX-License-Identifier: GPL-2.0-or-later
+// Copyright The XCSoar Project
+
+#pragma once
+
+#include "Validity.hpp"
+#include "time/Stamp.hpp"
+#include "Atmosphere/Temperature.hpp"
+
+/** Information about the engine sensor box values,
+ *  temperatures and revolutions per second
+*/
+struct EngineState
+{
+  Validity sensbox_flags_available;
+  uint8_t sensbox_flags;
+
+  Validity revs_per_sec_available;
+  uint16_t revs_per_sec;
+
+  Validity temp1_available;
+  Temperature temp1;
+
+  Validity temp2_available;
+  Temperature temp2;
+
+  void Clear() noexcept{
+    sensbox_flags = 0;
+    sensbox_flags_available.Clear();
+    revs_per_sec = 0;
+    revs_per_sec_available.Clear();
+    temp1_available.Clear();
+    temp2_available.Clear();
+  }
+
+  void Reset() noexcept{
+    Clear();
+  }
+
+  void Expire(TimeStamp clock, std::chrono::steady_clock::duration max_age) noexcept {
```

Comment:
Why is max_age a parameter? In all other structs, this is hard-coded. And there is just one caller, which hard-codes the value.

Reply:
btw. do you know what 3 seconds mean? Is this value your decision, or did you just copy it from somewhere else?